src/java.base/share/classes/com/sun/crypto/provider/DHPublicKey.java

```diff
@@ -339,11 +368,32 @@ private void readObject(ObjectInputStream stream)
         if ((key == null) || (key.length == 0)) {
             throw new InvalidObjectException("key not deserializable");
         }
-        this.key = key.clone();
         if ((encodedKey == null) || (encodedKey.length == 0)) {
             throw new InvalidObjectException(
                     "encoded key not deserializable");
         }
-        this.encodedKey = encodedKey.clone();
+        // check if the "encodedKey" value matches the deserialized fields
+        DHComponents c;
+        byte[] encodedKeyIntern = encodedKey.clone();
+        try {
+            c = decode(encodedKeyIntern);
+        } catch (IOException e) {
+            InvalidObjectException ioe = new InvalidObjectException("Invalid encoding");
+            if (ioe != null) {
+                ioe.initCause(e);
+            }
```

Comment:
The null check is redundant.
```suggestion
            ioe.initCause(e);
```

Reply:
Removed.